fees/first-crypto-bank/index.ts

```diff
@@ -0,0 +1,104 @@
+import * as sdk from "@defillama/sdk";
+import { getBalance } from "@defillama/sdk/build/eth";
+import { Adapter, ChainBlocks, FetchResultFees } from "../../adapters/types";
+import { getBlock } from "../../helpers/getBlock";
+import { getPrices } from "../../utils/prices";
+import { CHAIN } from "../../helpers/chains";
+import postgres from "postgres";
+
+/** Address to check = bankpadtreasury.eth */
+const CONTRACT_ADDRESS = "0x67262A61c0A459Fff172c22E60DBC730393BF790";
+
+/** Check ether balance at a given block number*/
+interface IBalance {
+    daily: string;
+    total: string;
+}
+
+const checkBalance = async (timestamp: number, chainBlocks: ChainBlocks): Promise<IBalance> => {
+    const fromTimestamp = timestamp - 60 * 60 * 24
+    const toTimestamp = timestamp
+    const currentBlock = await getBlock(toTimestamp, CHAIN.ETHEREUM, chainBlocks);
+    const yesterdayBlock = await getBlock(fromTimestamp, CHAIN.ETHEREUM, {});
+
+    const paramsCurrent = {
+        target: CONTRACT_ADDRESS,
+        block: currentBlock,
+        chain: CHAIN.ETHEREUM,
+    }
+
+    const paramsYesterday = {
+        target: CONTRACT_ADDRESS,
+        block: yesterdayBlock,
+        chain: CHAIN.ETHEREUM,
+    }
+
+    const currentBalance = (await getBalance(paramsCurrent)).output;
+    const yesterdayBalance = (await getBalance(paramsYesterday)).output;
+
+    return {
+        daily: (Number(currentBalance) - Number(yesterdayBalance)).toString(),
+        total: Number(currentBalance).toString()
+    } as IBalance
+}
+interface IData {
+    eth_value: string;
+  }
+
+/** Calculate USD equivalent for a given ether amount */
+async function usdEquivalent(_: string, timestamp: number) {
+    const sql = postgres(process.env.INDEXA_DB!);
+
+    const now = new Date(timestamp * 1e3)
+    const dayAgo = new Date(now.getTime() - 1000 * 60 * 60 * 24)
+    try {
+      const revenue_split = await sql`
+        SELECT
+          block_number,
+          block_time,
+          "value" / 1e18 as eth_value,
+          encode(transaction_hash, 'hex') AS HASH,
+          encode(to_address, 'hex') AS to_address
+        FROM
+          ethereum.traces
+        WHERE
+          block_number > 17539904
+          and to_address = '\\x67262A61c0A459Fff172c22E60DBC730393BF790'
```

Comment:
could you check this address, i check it is a wellet not contract there is not trace, i revice endpoint for this protocol not sure is it  correct `https://api.firstcryptobank.capital/reports/revenue_by_day`

Reply:
@dtmkeng can forget about endpoint... adapter is good. address is where revenue is collected